From ee0f50ec731b5d65367d845b0bd1acb6ae0c903b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:36 -0400
Subject: [powerpc] FA_DUMP depends on KEXEC

Message-id: <1410545655-205645-168-git-send-email-dzickus@redhat.com>
Patchwork-id: 94174
O-Subject: [RHEL7 PATCH 167/626] powerpc: FA_DUMP depends on KEXEC
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 90890b1ef568a54547ef0490a7c15d7357b98bc9
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Mon Oct 28 15:00:35 2013 +1100

    powerpc: FA_DUMP depends on KEXEC

    If you try and build the FA_DUMP code with CONFIG_KEXEC=n, you see
    errors such as the following:

    arch/powerpc/kernel/fadump.c
      408:2: error: 'crashing_cpu' undeclared (first use in this function)
      410:2: error: implicit declaration of function 'crash_save_vmcoreinfo'
      513:22: error: storage size of 'prstatus' isn't known
      520:2: error: implicit declaration of function 'elf_core_copy_kernel_regs'
      521:36: error: 'KEXEC_CORE_NOTE_NAME' undeclared (first use in this function)
      624:49: error: 'note_buf_t' undeclared (first use in this function)
      872:2: error: implicit declaration of function 'paddr_vmcoreinfo_note'
      874:18: error: 'vmcoreinfo_max_size' undeclared (first use in this function)

    This is because although FA_DUMP doesn't use kexec as the actual reboot
    mechanism, it does use parts of the kexec code to assemble/disassemble
    the crash image.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index f90906c..940666e 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -411,7 +411,7 @@ config CRASH_DUMP
 
 config FA_DUMP
  bool "Firmware-assisted dump"
- depends on PPC64 && PPC_RTAS && CRASH_DUMP
+ depends on PPC64 && PPC_RTAS && CRASH_DUMP && KEXEC
  help
    A robust mechanism to get reliable kernel crash dump with
    assistance from firmware. This approach does not use kexec,
-- 
1.7.1